src/shell.js

```diff
@@ -195,7 +195,12 @@ if (ENVIRONMENT_IS_NODE) {
 #endif
 
 #if NODEJS_CATCH_REJECTION
-  process['on']('unhandledRejection', abort);
+  // Without this older versions of node (< v15) will log unhandled rejections
+  // but return 0, which is not normally the desired behaviour.  This is
+  // not be needed with node v15 and about because it is now the default
+  // behaviour:
+  // See https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode
+  process['on']('unhandledRejection', function(reason) { throw reason; });
```

Comment:
Is this fix here that `abort()` modifies the input object before throwing it again? If so, perhaps we can modify `abort` to not do that?

Reply:
`abort` doesn't expect an exception as arg 0.  It will always generate a new exception and a new stack trace based on where its called from:
https://github.com/emscripten-core/emscripten/blob/f782b50a7f8dded7cd0e2c7ee4fed41ab743f5c0/src/preamble.js#L605

I think that is WAI for abort()... one expects the location of the call to abort to be the top of the stacktrace.

Here we just want to mimick the behaviour of node v15 and above I believe which is to treat unhandled rejection as a normal exception (and let `uncaughtException` decide what to do about it.

Reply:
Ah yes, I forgot about that line you quoted, good point.

I wonder if we can take the stack trace from a given exception and graft it onto that one, but that could be a separate issue I guess.

Overall, the lesson here is not to use `abort` as a rethrow, and I guess this PR does exactly that in that one place (and I can't think offhand of other places that exist).